src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -27,32 +27,21 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
   options.reconnectionHandler = options.reconnectionHandler || window['Blazor'].defaultReconnectionHandler;
   logger.log(LogLevel.Information, 'Starting up blazor server-side application.');
 
-  // Initialize statefully prerendered circuits and their components
-  // Note: This will all be removed soon
   const initialConnection = await initializeConnection(options, logger);
-  const circuits = discoverPrerenderedCircuits(document);
-  for (let i = 0; i < circuits.length; i++) {
-    const circuit = circuits[i];
-    for (let j = 0; j < circuit.components.length; j++) {
-      const component = circuit.components[j];
-      component.initialize();
-    }
-  }
-
   const circuit = await startCircuit(initialConnection);
   if (!circuit) {
-    logger.log(LogLevel.Information, 'No preregistered components to render.');
+    throw new Error('Ciruit failed to start.');
   }
 
   const reconnect = async (existingConnection?: signalR.HubConnection): Promise<boolean> => {
     if (renderingFailed) {
       // We can't reconnect after a failure, so exit early.
       return false;
     }
-    const reconnection = existingConnection || await initializeConnection(options, logger);
-    const results = await Promise.all(circuits.map(circuit => circuit.reconnect(reconnection)));
 
-    if (reconnectionFailed(results)) {
+    const reconnection = existingConnection || await initializeConnection(options, logger);
+    if (!(await circuit.reconnect(reconnection))) {
+      logger.log(LogLevel.Information, 'Reconnection attempt failed.');
```

Comment:
This seemed worth to add to logging. Please double-check me on all of the .ts changes, I'm not super familiar with the organization of the codez.

Reply:
This looks good. I'm extremely keen to reorganize most of the code that starts in `Boot.Server.ts` as it's grown into a pretty weird shape over time. Now that one of the major sources of complexity is being removed, I think there's an opportunity to make this much more approachable.

Not planning to actually do anything on this pre-3.0 though, as it will just be pure refactoring and not touch any public API or behavior.